### DISCUSSION TITLE
new kotlin and xml

### DISCUSSION THREAD

**Discussion by morganbr12** [community/community#138379] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

what are your thoughts in .kt and .xml

**Comment by itisrude** [comment, marked as answer]
Hey morganbr12,

Kotlin (`.kt`) and XML are widely used together in Android development, and they complement each other well for structuring the backend logic and frontend UI. Here are my thoughts:

- Kotlin: It's modern, concise, and much easier to read compared to Java. It also handles null safety well, which can help avoid a lot of runtime errors. The language's expressiveness allows for faster development and easier maintenance.

- XML: Still a solid choice for defining UI layouts in Android, although Jetpack Compose (which replaces XML) is gaining traction. XML is great for clear separation of design and logic but can get a bit cumbersome for complex UIs.

If you’re starting fresh, Kotlin with Jetpack Compose might be worth looking into since it's Google’s recommended modern toolkit for building UIs.

Let me know if you want more insights on this!

**Reply by morganbr12** [reply]
thank you very much for your explanation. it means much to me
